components/test_pd_client/src/pd.rs

```diff
@@ -1392,6 +1392,17 @@ impl TestPdClient {
         panic!("region {} must have leader: {:?}", region_id, peer);
     }
 
+    pub fn region_leader_must_not_be(&self, region_id: u64, peer: metapb::Peer) {
+        for _ in 0..500 {
+            sleep_ms(10);
+            if let Some(p) = self.cluster.rl().leaders.get(&region_id) {
+                if *p == peer {
+                    panic!("region {} must not have leader: {:?}", region_id, peer);
+                }
```

Comment:
```suggestion
                }
                return;
```
Once there has a vaild peer, should return instead of continue the loop.

Reply:
I initially thought this loop, with its sleep function, could be used to wait for the async operation (e.g., transfer_leader, not must_transfer_leader) to complete and then check the region leader. However, this approach was confusing, so I switched to using sleep + check_region_leader.